apis/v1alpha2/httproute_types.go

```diff
@@ -336,18 +319,38 @@ type HTTPHeaderMatch struct {
 	// +kubebuilder:default=Exact
 	Type *HeaderMatchType `json:"type,omitempty"`
 
-	// Values is a map of HTTP Headers to be matched.
-	// It MUST contain at least one entry.
+	// Name is the name of the HTTP Header to be matched. Name matching MUST be
+	// case insensitive. (See
+	// https://tools.ietf.org/html/rfc7230#section-3.2).
 	//
-	// The HTTP header field name to match is the map key, and the
-	// value of the HTTP header is the map value. HTTP header field name matching
-	// MUST be case-insensitive.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"Name"`
+
+	// Value is the value of HTTP Header to be matched.
 	//
-	// Multiple match values are ANDed together, meaning, a request
-	// must match all the specified headers to select the route.
-	Values map[string]string `json:"values"`
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=4096
```

Comment:
4096 should be good enough for most cases but someone might come in future wanting more. We can revisit then.
People use HTTP headers for all sorts of use-cases.

Reply:
Agree that we can likely raise this if needed. This was roughly based on the lowest default limit among relatively common implementations, but of course everything is configurable: https://stackoverflow.com/a/8623061.